modules/gatsby/src/gatsby-node/index.js

```diff
@@ -2,54 +2,115 @@ const {log, COLOR} = require('../utils/log');
 
 const onCreateWebpackConfig = require('./create-webpack-config');
 const createPages = require('./create-pages');
-const {processNewMarkdownNode, cleanupMarkdownNode, addSiblingNodes} = require('./process-nodes-markdown');
+const {
+  processNewMarkdownNode,
+  cleanupMarkdownNode,
+  addSiblingNodes
+} = require('./process-nodes-markdown');
 const {processNewDocsJsonNode} = require('./process-nodes-json');
 
 // TODO/ib - avoid globals
 const docNodes = {};
 let tocNode = null;
 
-function onCreateNode({ node, actions, getNode }) {
+function onCreateNode({node, actions, getNode}) {
   // log.log({color: COLOR.CYAN}, `Processed node`)();
 
   // Add missing fields to markdown nodes
-  cleanupMarkdownNode({ node, actions, getNode });
+  cleanupMarkdownNode({node, actions, getNode});
 
   switch (node.internal.type) {
-  case "MarkdownRemark":
-    // Note: MarkdownRemark nodes are created by the gatsby-transformer-remark
-    // markdown parser. These are different from the original file nodes
-    // for the markdown files created by the gatsby-source-filesystem plugin.
-    processNewMarkdownNode({ node, actions, getNode }, docNodes, tocNode);
-    break;
-
-  case 'DocsJson':
-    tocNode = processNewDocsJsonNode({ node, actions, getNode }, docNodes);
-    break;
-
-  default:
+    case 'MarkdownRemark':
+      // Note: MarkdownRemark nodes are created by the gatsby-transformer-remark
+      // markdown parser. These are different from the original file nodes
+      // for the markdown files created by the gatsby-source-filesystem plugin.
+      processNewMarkdownNode({node, actions, getNode}, docNodes, tocNode);
+      break;
+
+    case 'DocsJson':
+      tocNode = processNewDocsJsonNode({node, actions, getNode}, docNodes);
+      break;
+
+    default:
   }
 }
 
-function setFieldsOnGraphQLNodeType({ type, actions }) {
-  const { name } = type;
-  const { createNodeField } = actions;
-  if (name === "MarkdownRemark") {
+function setFieldsOnGraphQLNodeType({type, actions}) {
+  const {name} = type;
+  const {createNodeField} = actions;
+  if (name === 'MarkdownRemark') {
     addSiblingNodes(createNodeField);
   }
-};
+}
+
+function sourceNodes({actions}) {
+  const {createTypes} = actions;
```

Comment:
I recommend keeping this file small and focused on defining the gatsby callbacks, and delegating to functions in other files for actual processing.

I.e. create a new file `define-graphql-schema.js` or similar and move the implementation there.